src/Appwrite/Platform/Action.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Appwrite\Platform;
+
+use Swoole\Coroutine as Co;
+use Utopia\Database\Database;
+use Utopia\Database\Query;
+use Utopia\Platform\Action as UtopiaAction;
+
+class Action extends UtopiaAction
+{
+    /**
+     * Log Error Callback
+     *
+     * @var callable
+     */
+    protected mixed $logError;
+
+    /**
+     * Foreach Document
+     * Call provided callback for each document in the collection
+     *
+     * @param string $projectId
+     * @param string $collection
+     * @param array $queries
+     * @param callable $callback
+     *
+     * @return void
+     */
+    protected function foreachDocument(Database $database, string $collection, array $queries = [], callable $callback = null, int $limit = 1000, bool $concurrent = false): void
```

Comment:
Added a Discord thread about moving this function with proper tests to the database instance.